lib/sqlalchemy/sql/elements.py

```diff
@@ -3412,6 +3414,23 @@ def wrapped_column_expression(self):
         return self.clause
 
 
+
+class TryCast(Cast):
+    """Represent a TRY_CAST expression.
+
+    Details on :class:`.TryCast` usage is at :func:`.try_cast`.
+
+    .. seealso::
+
+        :func:`.try_cast`
+
+        :ref:`tutorial_casts`
+    """
+
+    __visit_name__ = "try_cast"
+    inherit_cache = True
```

Comment:
I just blindly copied these attributes, I don't understand their significance. And I removed `stringify_dialect = "mssql"`, which seems like that probably needs to go somewhere else instead...

Reply:
That would implement it for ALL backends, even ones that don't support TRY_CAST(), right? and that is fine, the inevitable crash will happen in the right place?

Reply:
no, StrSQLCompiler is a special compiler that only comes up when you do `str(stmt)`, so that people can see a stringify.  it's not the "main"  compiler that's used for execute().

Reply:
OK, I think that makes sense. Moved in newest version.

Reply:
@zzzeek would still love confirmation that I was correct with removing the `stringify_dialect = "mssql"`